File: Projeto_Final/FilmoTeca/app/models/Usuario.py
from pydantic import BaseModel, Field, validator
from typing import List, Optional
from app.models.Filme import Filme

class Usuario(BaseModel):
    id: Optional[int] = Field(default=None)
    nome: str
    email: str
    senha: str
    favoritos: Optional[List[Filme]] = []

File: Projeto_5/Projeto 5 POO/src/Usuario.cpp
#include "Usuario.h"

Usuario::Usuario() : nome(""), email(""), tipo("") {}

Usuario::Usuario(const std::string& nome, const std::string& email, const std::string& tipo)
    : nome(nome), email(email), tipo(tipo) {}

Usuario::~Usuario() {}

const std::string& Usuario::getNome() const {
    return nome;
}

const std::string& Usuario::getEmail() const {
    return email;
}

const std::string& Usuario::getTipo() const {
    return tipo;
}
